plugins/auto-sizes/optimization-detective.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Optimization Detective extensions by Auto Sizes.
+ *
+ * @since n.e.x.t
+ * @package auto-sizes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Visits responsive lazy-loaded IMG tags to ensure they include sizes=auto.
+ *
+ * @since n.e.x.t
+ *
+ * @param OD_Tag_Visitor_Context $context Tag visitor context.
+ * @return bool Whether visited.
+ */
+function auto_sizes_visit_tag( OD_Tag_Visitor_Context $context ): bool {
+	if ( 'IMG' !== $context->processor->get_tag() ) {
+		return false;
+	}
+
+	$sizes = $context->processor->get_attribute( 'sizes' );
+	if ( ! is_string( $sizes ) || 'lazy' !== $context->processor->get_attribute( 'loading' ) ) {
+		return false;
+	}
+
+	$sizes = preg_split( '/\s*,\s*/', $sizes );
+	if ( is_array( $sizes ) && ! in_array( 'auto', $sizes, true ) ) {
+		array_unshift( $sizes, 'auto' );
+		$context->processor->set_attribute( 'sizes', join( ', ', $sizes ) );
+	}
+
+	return true;
```

Comment:
In reality this can return `false` per #1342.

Reply:
Happy to defer to whatever you think here.

Reply:
```suggestion
	return false; // Since this tag visitor does not require this tag to be included in the URL Metrics.
```